modules/compiler/builtins/abacus/include/abacus/internal/is_odd.h

```diff
@@ -45,13 +45,17 @@ inline typename TypeTraits<T>::SignedType is_odd(const T &x) {
   const SignedType unbiasedExp = abacus::internal::logb_unsafe(x);
 
   // We're using the exponent to shift, we can only shift by [0, <bits in type>)
-  const SignedType validExp = (unbiasedExp >= 0) & (unbiasedExp < numBits);
+  // We will use validExp to mask, so need -1 for true even for scalars.
```

Comment:
Could we add a little more comments here - I'm not quite sure i follow the logic. Also scalar and vector look very different - which confuses me based on your description, which suggested they followed similar shift logic.

Reply:
Relational operators are different for scalar and vector: for scalar, they return 0 for false, 1 for true, for vectors they return 0 for false, -1 for true. This change makes sure they return 0 for false, -1 for true, both for scalar and for vector, so that the rest of the code works the same for scalar and vector alike. We need -1 (all bits 1) for true because otherwise when we do `validExp & unbiasedExp`, we get the wrong value. Will have a think about how to phrase this better in the comment.

Reply:
Should hopefully be clearer now, and also expanded the comment on `multiply_exact`.